app/state_machines/transaction_process_state_machine.rb

```diff
@@ -76,13 +82,29 @@ class TransactionProcessStateMachine
     send_new_transaction_email(transaction) if transaction.conversation.payment?
   end
 
+  # "guard_transition" is before SQL BEGIN-COMMIT block
+  # instead, "before_transition" is inside the block
+  before_transition(to: :preauthorized) do |transaction, transition|
+    validate_before_preauthorized(transaction, transition)
+  end
+
+  before_transition(to: :payment_intent_requires_action) do |transaction, transition|
```

Comment:
@bladealslayer 
Oops, now it validates both.

Reply:
👍 I guess we also need one:

```ruby
after_transition_failure(to: :payment_intent_requires_action) do |transaction|
    void_payment(transaction)
  end
```

right, @aivils?